coil-default/src/main/java/coil/Coil.kt

```diff
@@ -26,29 +26,29 @@ object Coil {
      * Get the default [ImageLoader]. Creates a new instance if none has been set.
      */
     @JvmStatic
-    fun imageLoader(context: Context): ImageLoader = imageLoader ?: buildImageLoader(context)
+    fun imageLoader(context: Context): ImageLoader = imageLoader ?: newImageLoader(context)
 
     /**
      * Set the default [ImageLoader]. Shutdown the current instance if there is one.
      */
     @JvmStatic
     fun setImageLoader(loader: ImageLoader) {
-        setImageLoader(object : ImageLoaderProvider {
-            override fun getImageLoader() = loader
+        setImageLoader(object : ImageLoaderFactory {
+            override fun newImageLoader() = loader
         })
     }
 
     /**
-     * Set a lazy callback to create the default [ImageLoader]. Shutdown the current instance if there is one.
-     * The [provider] is guaranteed to be called at most once.
+     * Set the [ImageLoaderFactory] that will be used to create the default [ImageLoader].
+     * Shutdown the current instance if there is one. The [factory] is guaranteed to be called at most once.
      *
-     * Using this method to set an explicit [provider] takes precedence over an [Application] that
-     * implements [ImageLoaderProvider].
+     * Using this method to set an explicit [factory] takes precedence over an [Application] that
+     * implements [ImageLoaderFactory].
      */
     @JvmStatic
     @Synchronized
-    fun setImageLoader(provider: ImageLoaderProvider) {
-        imageLoaderProvider = provider
+    fun setImageLoader(factory: ImageLoaderFactory) {
```

Comment:
Should the method be `setImageLoaderFactory` instead?

Reply:
Actually, I get it now. Probably best to keep it like this so the consumer gets a choice of how they want to provide the image loader 👍

Reply:
Yep, I think it'll make more sense too when we can make `ImageLoaderFactory` a `fun interface` in Kotlin 1.4 so it's callable using `Coil.setImageLoader { ImageLoader(context) }`. Mostly I wanted to give both methods the same name to show they overwrite each other.